.github/workflows/push-pr.yaml

```diff
@@ -0,0 +1,77 @@
+name: Push/PR
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+
+jobs:
+  deps:
+    name: Check dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: Ensure deps are clean
+        run: |
+          make deps
+          ./scripts/enforce-clean
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Needed for version script to work.
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: Build
+        run: |
+          make build
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+      - name: Build container
+        uses: docker/build-push-action@v6
+        with:
+          push: false
+          platforms: linux/amd64,linux/arm64
+          # Needed for this action to use local context incl. binaries built in the previous step.
+          # https://github.com/docker/build-push-action?tab=readme-ov-file#path-context
+          # TODO: Make docker build self-contained.
+          context: .
+          tags: ci.local/${{ github.repository }}
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: Test
+        run: |
+          make build-xk6-linux-amd64 # Tests require a valid k6 binary in dist/k6
+          make test
+
+  golangci:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Get golangci-lint version that `make lint` uses.
+      - id: version
+        run: |-
+          echo "golangci=$(make golangci-lint-version)" >> $GITHUB_OUTPUT
+      # Use the golangci-lint action, which provides Github-specific features such as diff annotations.
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: ${{ steps.version.outputs.golangci }}
+          args: --timeout=5m # 1m is not enough, experimentally.
+          only-new-issues: true
```

Comment:
I do have a problem with this (ignoring the obvious bug): this doesn't match what `make lint` does. That's my biggest issue with CI/CD system: what the remote system does does not match what the developer sees locally, and that ends up in that nasty cycle of "fix it" / "push" / "try again" / "push" / ...

The (maybe not) obvious bug is that golangci-lint's configuration for this repository is located in `scripts/configs/golangci.yml`. That probably should be moved to `.golangci.yaml` because that has a couple of advantages.

I'm open to reading arguments in favor of using a GHA, but those have to be weighted against developer's sanity.

Reply:
> The (maybe not) obvious bug is that golangci-lint's configuration for this repository is located in scripts/configs/golangci.yml. That probably should be moved to .golangci.yaml because that has a couple of advantages.

That's a good point. I'll take this chance to move it to the default location so other things can also pick it up.

> I'm open to reading arguments in favor of using a GHA, but those have to be weighted against developer's sanity.

I couldn't agree more! I'm all in for matching CI and local behavior (that's why among other things I push for self-contained Docker builds), but my definition of developer sanity also includes not having to go back and forth from `line:col` entries in the logs to the source code. Diff annotations are very convenient, and that's why I prefer using GHA-native things when available.

I'll add the `make lint` step to the pipeline too, so we don't lose linters. We'd be running golangci twice, but I think that's worth it.